src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -80,12 +79,18 @@ protected virtual HttpWebRequest CreateWebRequest(RequestData requestData)
 			request.ReadWriteTimeout = timeout;
 
 			//WebRequest won't send Content-Length: 0 for empty bodies
-			//which goes against RFC's and might break i.e IIS hen used as a proxy.
-			//see: https://github.com/elasticsearch/elasticsearch-net/issues/562
-			var m = requestData.Method.GetStringValue();
-			request.Method = m;
-			if (m != "HEAD" && m != "GET" && (requestData.PostData == null))
-				request.ContentLength = 0;
+			//which goes against RFC's and might break i.e IIS when used as a proxy.
+			//see: https://github.com/elasticsearch/elasticsearch-net/issues/562.
+			//Transfer-Encoding: chunked must be set for POST and PUT where there is a body.
+			var method = requestData.Method.GetStringValue();
+			request.Method = method;
+			if (method != "HEAD" && method != "GET")
+			{
+				if (requestData.PostData == null)
+					request.ContentLength = 0;
+				else

```

Comment:
Does this need [SupportsTransferEncodingChunked](https://msdn.microsoft.com/en-us/library/system.net.http.httpclienthandlerextensions.supportstransferencodingchunked(v=vs.118).aspx) and if so should `AllowWriteStreamBuffering = false` depend on that too?